src/redux/sagas/expenditures/createStakedExpenditure.ts

```diff
@@ -71,69 +87,81 @@ function* createStakedExpenditure({
+        ready: false,
+      });
+
+      yield createGroupTransaction(deposit, batchKey, meta, {
+        context: ClientType.TokenLockingClient,
+        methodName: 'deposit(address,uint256,bool)',
+        identifier: colonyAddress,
+        params: [tokenAddress, missingActiveTokens, false],
+        ready: false,
+      });
+
+      yield takeFrom(approve.channel, ActionTypes.TRANSACTION_CREATED);
+
+      yield takeFrom(deposit.channel, ActionTypes.TRANSACTION_CREATED);
+
+      yield initiateTransaction({ id: approve.id });
+
+      yield waitForTxResult(approve.channel);
+
+      yield initiateTransaction({ id: deposit.id });
+
+      yield waitForTxResult(deposit.channel);
+    }
+
+    yield createGroupTransaction(approveStake, batchKey, meta, {
       context: ClientType.ColonyClient,
       methodName: 'approveStake',
       identifier: colonyAddress,
       params: [stakedExpenditureAddress, createdInDomain.nativeId, stakeAmount],
-      group: {
-        key: batchKey,
-        id: meta.id,
-        index: 0,
-      },
       ready: false,
     });
 
-    yield fork(createTransaction, makeExpenditure.id, {
+    yield createGroupTransaction(makeExpenditure, batchKey, meta, {
       context: ClientType.StakedExpenditureClient,
       methodName: 'makeExpenditureWithStake',
       identifier: colonyAddress,
-      group: {
-        key: batchKey,
-        id: meta.id,
-        index: 1,
-      },
       ready: false,
     });
 
-    yield fork(createTransaction, setExpenditureValues.id, {
+    yield createGroupTransaction(setExpenditureValues, batchKey, meta, {
       context: ClientType.ColonyClient,
       methodName: 'multicall',
       identifier: colonyAddress,
-      group: {
-        key: batchKey,
-        id: meta.id,
-        index: 2,
-      },
       ready: false,
     });
 
     if (isStaged) {
-      yield fork(createTransaction, setExpenditureStaged.id, {
+      yield createGroupTransaction(setExpenditureStaged, batchKey, meta, {
         context: ClientType.StagedExpenditureClient,
         methodName: 'setExpenditureStaged',
         identifier: colonyAddress,
-        group: {
-          key: batchKey,
-          id: meta.id,
-          index: 3,
-        },
         ready: false,
       });
     }
 
     if (annotationMessage) {
-      yield fork(createTransaction, annotateMakeStagedExpenditure.id, {
-        context: ClientType.ColonyClient,
-        methodName: 'annotateTransaction',
-        identifier: colonyAddress,
-        group: {
-          key: batchKey,
-          id: meta.id,
-          index: isStaged ? 4 : 3,
+      yield createGroupTransaction(
+        annotateMakeStagedExpenditure,
+        batchKey,
+        meta,
+        {
+          context: ClientType.ColonyClient,
+          methodName: 'annotateTransaction',
+          identifier: colonyAddress,
+          ready: false,
         },
-        ready: false,
-      });
+      );
     }
 
```

Comment:
This is where I think you can initiate the first transaction, once all transactions in the group have been created.

Reply:
Ohhh all right I shall give that a go!

Reply:
My changes should appear a couple of lines beneath this comment 👇